trio/_core/_tests/test_multierror.py

```diff
@@ -362,16 +369,16 @@ def catch_RuntimeError(exc):
 @pytest.mark.skipif(
     sys.implementation.name != "cpython", reason="Only makes sense with refcounting GC"
 )
-def test_MultiError_catch_doesnt_create_cyclic_garbage():
+def test_MultiError_catch_doesnt_create_cyclic_garbage() -> None:
     # https://github.com/python-trio/trio/pull/2063
     gc.collect()
     old_flags = gc.get_debug()
 
-    def make_multi():
+    def make_multi() -> NoReturn:
         # make_tree creates cycles itself, so a simple
         raise MultiError([get_exc(raiser1), get_exc(raiser2)])
 
-    def simple_filter(exc):
+    def simple_filter(exc: BaseException) -> Exception | RuntimeError:
```

Comment:
yeah, I did it intentionally to mirror the function body. I realize now the proper way to do it would maybe be `(exc: T_Exception) -> T_Exception | RunTimeError`, but that's kinda overkill.